v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/model/Mod.java

```diff
@@ -0,0 +1,273 @@
+    return changeJson;
+  }
+
+  /**
+   * The seconds part of the timestamp at which the modifications within were committed in Cloud
+   * Bigtable.
+   */
+  public long getCommitTimestampSeconds() {
+    return commitTimestampSeconds;
+  }
+
+  /**
+   * The nanoseconds part of the timestamp at which the modifications within were committed in Cloud
+   * Bigtable.
+   */
+  public int getCommitTimestampNanos() {
+    return commitTimestampNanos;
+  }
+
+  /** The type of operation that caused the modifications within this record. */
+  public ModType getModType() {
+    return modType;
+  }
+
+  @Override
+  public boolean equals(@javax.annotation.Nullable Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Mod)) {
+      return false;
+    }
+    Mod that = (Mod) o;
+    return StringUtils.equals(changeJson, that.changeJson)
+        && commitTimestampSeconds == that.commitTimestampSeconds
+        && commitTimestampNanos == that.commitTimestampNanos
+        && modType == that.modType;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(changeJson, commitTimestampSeconds, commitTimestampNanos, modType);
+  }
+
+  @Override
+  public String toString() {
+    return "Mod{"
+        + "changeJson='"
+        + changeJson
+        + '\''
+        + ", commitTimestampSeconds="
+        + commitTimestampSeconds
+        + ", commitTimestampNanos="
+        + commitTimestampNanos
+        + ", modType="
+        + modType
+        + '}';
+  }
+
+  public String toJson() throws JsonProcessingException {
+    return new ObjectMapper().writeValueAsString(this);
+  }
+
+  private String encodeBytes(ByteString rowKey) {
+    if (rowKey == null) {
+      return null;
+    } else {
+      return Base64.getEncoder().encodeToString(rowKey.toByteArray());
+    }
+  }
+
+  private String cbtTimestampToPubSub(Instant timestamp) {
+    if (timestamp == null) {
+      return null;
+    }
+    if (timestamp.getEpochSecond() == 0 && timestamp.getNano() == 0) {
+      return TIMESTAMP_FORMATTER.get().format(java.time.Instant.EPOCH);
+    } else {
+      return TIMESTAMP_FORMATTER
+          .get()
+          .format(java.time.Instant.ofEpochSecond(timestamp.getEpochSecond(), timestamp.getNano()));
+    }
+  }
+
+  private String cbtTimestampMicrosToPubSubInt(Long timestampMicros) {
+    if (timestampMicros == null) {
+      return null;
+    }
+    return Long.toString(timestampMicros);
+  }
+
+  private String cbtTimestampMicrosToPubSub(Long timestampMicros) {
+    if (timestampMicros == null) {
+      return null;
+    }
+    if (timestampMicros == 0) {
+      return TIMESTAMP_FORMATTER.get().format(java.time.Instant.EPOCH);
+    } else {
+      long seconds = timestampMicros / 1000000;
+      return TIMESTAMP_FORMATTER
+          .get()
+          .format(
+              java.time.Instant.ofEpochSecond(
+                  seconds, (timestampMicros - seconds * 1000000) * 1000));
+    }
+  }
```

Comment:
This function will become unused if TIMESTAMP, TIMESTAMP_TO and TIMESTAMP_FROM are removed.